### PR TITLE
[MM-61508] Fixed errcheck issues in upload_test.go, [MM-61510] Fixed errcheck issues in user_test.go

### PR DESCRIPTION
**Pull request by jessiekahn**
[MM-61508] Fixed errcheck issues in upload_test.go
[MM-61510] Fixed errcheck issues in user_test.go

#### Summary
Fixed all errcheck issues reported by golangci-lint in server/channels/app/upload_test.go and server/channels/app/user_test.go

#### Ticket Link
Fixes  [#29126](https://github.com/mattermost/mattermost/issues/29126)
Jira:  [MM-61508](https://mattermost.atlassian.net/browse/MM-61508)

Fixes [#29128](https://github.com/mattermost/mattermost/issues/29128)
Jira: [MM-61510](https://mattermost.atlassian.net/browse/MM-61510)

#### Release Note

[MM-61508]: https://mattermost.atlassian.net/browse/MM-61508?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[MM-61510]: https://mattermost.atlassian.net/browse/MM-61510?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[MM-61508]: https://mattermost.atlassian.net/browse/MM-61508?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[MM-61510]: https://mattermost.atlassian.net/browse/MM-61510?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: server/channels/app/user_test.go [hunk 1]
```diff
@@ -1096,30 +1130,6 @@ func TestPermanentDeleteUser(t *testing.T) {
...
-			ChannelId: th.BasicChannel.Id,
-			UserId:    th.BasicUser.Id,
-			Message:   "Scheduled post 2",
-		},
-		ScheduledAt: model.GetMillis() + 1000000,
-	}
-
-	createdScheduledPost2, appErr := th.App.SaveScheduledPost(th.Context, scheduledPost2, "")
-	require.Nil(t, appErr)
-
```
[the remark this answers is not in the document]
Comment (jessiekahn) [reply server/channels/app/user_test.go]: Added these back- sorry, I'm not sure how this got deleted

File: server/channels/app/user_test.go [hunk 2]
```diff
@@ -1139,15 +1149,6 @@ func TestPermanentDeleteUser(t *testing.T) {
...
 	require.False(t, exists, "Profile image wasn't deleted. err=%v", err)
-
-	// verify scheduled posts have been deleted
-	fetchedScheduledPost, scheduledPostErr := th.App.Srv().Store().ScheduledPost().Get(createdScheduledPost1.Id)
-	require.ErrorIs(t, scheduledPostErr, sql.ErrNoRows)
-	require.Nil(t, fetchedScheduledPost)
-
-	fetchedScheduledPost, scheduledPostErr = th.App.Srv().Store().ScheduledPost().Get(createdScheduledPost2.Id)
-	require.ErrorIs(t, scheduledPostErr, sql.ErrNoRows)
-	require.Nil(t, fetchedScheduledPost)
```
[the remark this answers is not in the document]
Comment (jessiekahn) [reply server/channels/app/user_test.go]: Added these back- sorry, I'm not sure how this got deleted

File: server/channels/app/user_test.go [hunk 3]
```diff
@@ -1095,7 +1129,7 @@ func TestPermanentDeleteUser(t *testing.T) {
 	err1 = sqlStore.GetMasterX().Select(&bots2, "SELECT * FROM Bots")
 	assert.NoError(t, err1)
 	assert.Equal(t, 0, len(bots2))
-
+
```
Comment (hanzei) [inline server/channels/app/user_test.go:1132]: Unrelated change?

File: server/channels/app/user_test.go [hunk 4]
```diff
@@ -1119,7 +1153,6 @@ func TestPermanentDeleteUser(t *testing.T) {
 
 	createdScheduledPost2, appErr := th.App.SaveScheduledPost(th.Context, scheduledPost2, "")
 	require.Nil(t, appErr)
-
```
Comment (hanzei) [inline server/channels/app/user_test.go:1122]: Unrelated change?

File: server/channels/app/user_test.go [hunk 5]
```diff
@@ -6,7 +6,6 @@ package app
 import (
 	"bytes"
 	"context"
-	"database/sql"
```
Comment (hanzei) [inline server/channels/app/user_test.go:9]: That seems wrong.

**Review by hanzei — CHANGES_REQUESTED** [review]
Nice, almost there. There are some minor issues with the diff.

File: server/channels/app/user_test.go (hunk 3 above)
Comment (jessiekahn) [reply server/channels/app/user_test.go:1132]: Not sure why this changed - I can't seem to see the change here should I take the blank line out?

File: server/channels/app/user_test.go (hunk 4 above)
Comment (jessiekahn) [reply server/channels/app/user_test.go:1122]: Not sure why this changed - Should I add a blank line here?

**Review by hanzei — APPROVED** [review]
LGTM :+1:

File: server/.golangci.yml [hunk 6]
```diff
@@ -129,8 +129,6 @@ issues:
         channels/app/team.go|\
         channels/app/team_test.go|\
         channels/app/upload.go|\
-        channels/app/upload_test.go|\
-        channels/app/user_test.go|\
```
Comment (rahimrahman) [inline server/.golangci.yml:133]: @hanzei learning question: would this cause the errcheck linter not to run on these 2 files?

**Review by rahimrahman — APPROVED** [review]
I only have just one question, but the rest LGTM!

File: server/.golangci.yml (hunk 6 above)
Comment (hanzei) [reply server/.golangci.yml:133]: The opposite is true. These changes remove the 2 files from the ignore list, allowing the linter to check them.

Comment (rahimrahman) [reply server/.golangci.yml:123]: Ah, would not have known that!  Thanks for the answer.
